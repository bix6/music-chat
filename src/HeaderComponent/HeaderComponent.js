import React from 'react';
import './HeaderComponent.css';

export default function HeaderComponent() {
    return (
        <header>
            <h1>Music Chat</h1>
        </header>
    );
}