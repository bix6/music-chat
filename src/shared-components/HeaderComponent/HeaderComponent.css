.header-component {
  text-align: center;
}

.header-component h1 {
  font-size: 4rem;
  font-family: monospace;
  margin: 1.5rem 0 0 0;
}
