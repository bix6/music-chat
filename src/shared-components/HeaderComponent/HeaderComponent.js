import React from 'react';
import { Link } from 'react-router-dom';

const headerStyles = {
    textAlign: 'center',
    margin: '2rem 0',
}

const h1Styles = {
    fontSize: '4rem',
    fontFamily: 'monospace'
}

export default function HeaderComponent() {
    return (
        <header style={ headerStyles }>
            <Link to="/">
                <h1 style={ h1Styles }>Music Chat</h1>
            </Link>
        </header>
    );
}