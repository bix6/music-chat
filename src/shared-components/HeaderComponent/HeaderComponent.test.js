import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter } from "react-router-dom";
import HeaderComponent from "./HeaderComponent";

describe("HeaderComponent Component", () => {
  it("Smoke Test: Renders Default", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <HeaderComponent />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Snapshot Test: Default", () => {
    const wrapper = shallow(<HeaderComponent />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
