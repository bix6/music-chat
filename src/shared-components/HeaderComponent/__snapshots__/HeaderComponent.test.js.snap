// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent Component Snapshot Test: Default 1`] = `
<header
  style={
    Object {
      "margin": "2rem 0",
      "textAlign": "center",
    }
  }
>
  <h1
    style={
      Object {
        "fontFamily": "monospace",
        "fontSize": "4rem",
      }
    }
  >
    Music Chat
  </h1>
</header>
`;
